### ISSUE TITLE
golang output plugins that multiple instances are configured lead to fluent-bit cannot be shut down gracefully

### ISSUE THREAD

**Issue by littlejoyo** [fluent/fluent-bit#6641]
## Bug Report

**Describe the bug**

When I configure multiple output instances and then want to shut down the service, the service cannot be shut down normally, and the SIGABRT signal is abnormal

go plugins link: https://github.com/fluent/fluent-bit-go

**To Reproduce**

- Example log message if applicable:( the last line is error)

```
Fluent Bit v2.0.5
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2023/01/04 11:06:00] [ info] [fluent bit] version=2.0.5, commit=, pid=1167856
[2023/01/04 11:06:00] [ info] [storage] ver=1.3.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2023/01/04 11:06:00] [ info] [cmetrics] version=0.5.7
[2023/01/04 11:06:00] [ info] [ctraces ] version=0.2.5
[2023/01/04 11:06:00] [ info] [input:cpu:cpu.0] initializing
[2023/01/04 11:06:00] [ info] [input:cpu:cpu.0] storage_strategy='memory' (memory only)
[2023/01/04 11:06:00] [ info] [input:dummy:dummy.1] initializing
[2023/01/04 11:06:00] [ info] [input:dummy:dummy.1] storage_strategy='memory' (memory only)
[flb-go] plugin parameter = ''
[flb-go] plugin parameter = ''
[2023/01/04 11:06:00] [ info] [sp] stream processor started
[0] cpu.local: [2023-01-04 11:06:01.071385046 +0800 CST, {"cpu2.p_user": 0, "cpu2.p_system": 0, "cpu3.p_cpu": 0, "cpu_p": 0, "user_p": 0, "system_p": 0, "cpu0.p_system": 0, "cpu1.p_system": 0, "cpu3.p_system": 0, "cpu0.p_cpu": 0, "cpu2.p_cpu": 0, "cpu0.p_user": 0, "cpu1.p_cpu": 0, "cpu1.p_user": 0, "cpu3.p_user": 0, }
[0] dummy.local: [2023-01-04 11:06:01.071495898 +0800 CST, {"message": [100 117 109 109 121], }
[0] cpu.local: [2023-01-04 11:06:02.071421947 +0800 CST, {"cpu0.p_cpu": 5, "cpu3.p_cpu": 5, "cpu3.p_system": 3, "system_p": 3.25, "cpu1.p_cpu": 5, "cpu1.p_user": 1, "cpu1.p_system": 4, "cpu2.p_cpu": 4, "cpu3.p_user": 2, "cpu_p": 4.75, "cpu0.p_system": 3, "cpu2.p_user": 1, "user_p": 1.5, "cpu0.p_user": 2, "cpu2.p_system": 3, }
[0] dummy.local: [2023-01-04 11:06:02.071474177 +0800 CST, {"message": [100 117 109 109 121], }
[0] cpu.local: [2023-01-04 11:06:03.071406778 +0800 CST, {"cpu0.p_cpu": 8, "cpu1.p_user": 4, "cpu1.p_system": 5, "cpu0.p_system": 5, "cpu2.p_cpu": 9, "cpu2.p_system": 5, "cpu3.p_system": 5, "cpu3.p_cpu": 8, "cpu3.p_user": 3, "cpu_p": 8.5, "system_p": 5, "cpu0.p_user": 3, "cpu1.p_cpu": 9, "cpu2.p_user": 4, "user_p": 3.5000000000000004, }
[0] dummy.local: [2023-01-04 11:06:03.071456445 +0800 CST, {"message": [100 117 109 109 121], }
[0] cpu.local: [2023-01-04 11:06:04.07142921 +0800 CST, {"system_p": 3, "cpu3.p_user": 3, "cpu3.p_system": 3, "cpu_p": 5, "cpu0.p_user": 2, "cpu2.p_system": 2, "user_p": 2, "cpu3.p_cpu": 6, "cpu0.p_cpu": 6, "cpu0.p_system": 4, "cpu1.p_cpu": 5, "cpu1.p_user": 1, "cpu1.p_system": 4, "cpu2.p_cpu": 4, "cpu2.p_user": 2, }
[0] dummy.local: [2023-01-04 11:06:04.07150042 +0800 CST, {"message": [100 117 109 109 121], }
^C[2023/01/04 11:06:05] [engine] caught signal (SIGINT)
[2023/01/04 11:06:05] [ warn] [engine] service will shutdown in max 5 seconds
[2023/01/04 11:06:05] [ info] [input] pausing cpu.0
[2023/01/04 11:06:05] [ info] [input] pausing dummy.1
[0] cpu.local: [2023-01-04 11:06:05.071454181 +0800 CST, {"cpu1.p_user": 3, "cpu1.p_system": 7, "cpu2.p_user": 3, "cpu2.p_system": 7, "cpu0.p_cpu": 11, "cpu1.p_cpu": 10, "cpu3.p_user": 3, "cpu_p": 10.25, "system_p": 7.000000000000001, "cpu2.p_cpu": 10, "user_p": 3.25, "cpu0.p_user": 4, "cpu0.p_system": 7, "cpu3.p_cpu": 10, "cpu3.p_system": 7, }
[0] dummy.local: [2023-01-04 11:06:05.071512488 +0800 CST, {"message": [100 117 109 109 121], }
[2023/01/04 11:06:06] [ info] [engine] service has stopped (0 pending tasks)
[2023/01/04 11:06:06] [engine] caught signal (SIGSEGV)
[2023/01/04 11:06:06] [ info] [input] pausing cpu.0
[2023/01/04 11:06:06] [ info] [input] pausing dummy.1
 **'fluent-bit -c ./fluent-bit.conf' terminated by signal SIGABRT (Abort)**
```
- Steps to reproduce the problem:

**Expected behavior**

 shut down the fluent-bit service gracefully,but it was not

**Screenshots**

**Your Environment**

* Version used:  
Fluent Bit v2.0.5
* Configuration:
```
[SERVICE]
    Flush        1
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf
    Plugins_File plugins.conf
    HTTP_Server  Off
    HTTP_Listen  0.0.0.0
    HTTP_Port    2020

[INPUT]
    Name cpu
    Tag  cpu.local
    Interval_Sec 1

[INPUT]
    Name dummy
    Tag  dummy.local

[OUTPUT]
    Name gstdout
    Match cpu*
    Id cpu_metrics

[OUTPUT]
    Name  gstdout
    Match dummy*
    Id dummy_metrics
```
* Environment name and version (e.g. Kubernetes? What version?):
native linux
* Operating System and version:
centos-8 
* Addition

If I only configure one output, it can shut down the service normally，the problem occurs in multiple instances！
hope to receive help！thank you!

**Comment by nokute78** [comment]
Could you test v2.0.7 or later version ?
It seems to be fixed from v2.0.7.

**Comment by littlejoyo** [comment]
> Could you test v2.0.7 or later version ?
> It seems to be fixed from v2.0.7.

I tried version 2.0.8 and it did solve the problem，thanks！
